public/assets/js/authFunctions.js

```diff
@@ -0,0 +1,28 @@
+function sendAuthRequest (channelName, password, url) {  // url === /signup or /login
+    return new Promise(function(resolve, reject) {
+        // make sure the claim name is still available
+        let xhttp;
+        const params = `username=${channelName}&password=${password}`;
+        console.log(params, url);
+        xhttp = new XMLHttpRequest();
```

Comment:
Is this the only place we're making XMLHttpRequests in the app? If so, I suppose this is okay, if not, we might want to standardize this a bit.